### DISCUSSION TITLE
Generating multiple suites with different configurations

### DISCUSSION THREAD

**Discussion by WisdomPill** [pytest-dev/pytest#12920] [Q&A]
I was looking for a way to launch all tests multiple times with different `DJANGO_SETTINGS_MODULE` is that possible?

specifically running all of [these tests](https://github.com/jazzband/django-redis/blob/2c7aae9c780fde3eebb5d96b4d9b56e29c563eb4/setup.cfg#L88),
if I am able to generate multiple tests were I can configure django on a function basis then it would be easier for me to parallelise it.
I tried using `pytest_configure` and it works for setting up django, but I would like to setup a different `DJANGO_SETTINGS_MODULE` and configure django for each different settings I have. At the moment the following works.

```python
from os import environ
from django import setup

def pytest_configure():
    environ["DJANGO_SETTINGS_MODULE"] = "settings.sqlite"
    setup()
```

So I started digging and trying out `pytest_generate_tests` since it looked like I could parametrize tests but I don't want to add the parameter to all of them, 
would be nice to use a fixture on function scope with autouse and configure it from there. 
So I tried it like the following but it looks like it is not used.

```python
@pytest.fixture(autouse=True, scope="function")
def django(settings_module) -> None:
    environ["DJANGO_SETTINGS_MODULE"] = settings_module
    setup()

def pytest_generate_tests(metafunc):
    settings_modules = [
        "settings.sqlite",
        "settings.sqlite_gzip"
    ]
    metafunc.parametrize("settings_module", settings_modules)
```

I am thinking it is not called early enough.

Could you help me understand how to implement it such solution if it's possible at all?

**Comment by dongfangtianyu** [comment]
Can an automatically running and parameterized fixture help you?

```python
@pytest.fixture(autouse=True, scope="function", params=["settings.sqlite", "settings.sqlite_herd"])
def django(request) -> None:
    environ["DJANGO_SETTINGS_MODULE"] = request.param
    setup()
```
More info:https://docs.pytest.org/en/stable/how-to/fixtures.html#fixture-parametrize
